horizon/facts/dependencies.py

```diff
@@ -28,3 +29,23 @@ def get_data_update_subscriber(
 DataUpdateSubscriberDependency = Annotated[
     DataUpdateSubscriber, Depends(get_data_update_subscriber)
 ]
+
+
+def get_wait_timeout(request: Request) -> float | None:
+    wait_timeout = request.headers.get(
+        "X-Wait-timeout", sidecar_config.LOCAL_FACTS_WAIT_TIMEOUT
+    )
+    try:
+        wait_timeout = float(wait_timeout)
+    except ValueError as e:
+        raise HTTPException(
+            status_code=400,
+            detail=f"Invalid X-Wait-timeout header, expected float, got {wait_timeout!r}",
+        ) from e
```

Comment:
this can be replaced with https://fastapi.tiangolo.com/tutorial/header-params/?h=header#header-parameters